packages/core/contracts/oracle/implementation/OptimisticOracle.sol

```diff
@@ -0,0 +1,389 @@
+        uint256 timestamp
+    );
+    event Settle(
+        address indexed requester,
+        address indexed proposer,
+        address indexed disputer,
+        bytes32 identifier,
+        uint256 timestamp,
+        int256 price,
+        uint256 payout
+    );
+
+    enum State {
+        Invalid, // Never requested.
+        Requested, // Requested, no other actions taken.
+        Proposed, // Proposed, but not expired or disputed yet.
+        Expired, // Proposed, not disputed, past liveness.
+        Disputed, // Disputed, but no DVM price returned yet.
+        Resolved, // Disputed and DVM price is available.
+        Settled // Final price has been set in the contract (can get here from Expired or Resolved).
+    }
+
+    struct Request {
+        address proposer;
+        address disputer;
+        IERC20 currency;
+        bool settled;
+        bool refundOnDispute;
+        int256 proposedPrice;
+        int256 resolvedPrice;
+        uint256 expirationTime;
+        uint256 reward;
+        uint256 finalFee;
+        uint256 bond;
+        uint256 customLiveness;
+    }
+
+    mapping(bytes32 => Request) public requests;
+
+    FinderInterface public finder;
+    uint256 public defaultLiveness;
+
+    constructor(
+        uint256 _liveness,
+        address _finderAddress,
+        address _timerAddress
+    ) public Testable(_timerAddress) {
+        finder = FinderInterface(_finderAddress);
+        _validateLiveness(_liveness);
+        defaultLiveness = _liveness;
+    }
+
+    function requestPrice(
+        bytes32 identifier,
+        uint256 timestamp,
+        IERC20 currency,
+        uint256 reward
+    ) external nonReentrant() returns (uint256 totalBond) {
+        require(getState(msg.sender, identifier, timestamp) == State.Invalid, "requestPrice: Invalid");
+        require(_getIdentifierWhitelist().isIdentifierSupported(identifier), "Unsupported identifier");
+        require(_getCollateralWhitelist().isOnWhitelist(address(currency)), "Unsupported currency");
+        uint256 finalFee = _getStore().computeFinalFee(address(currency)).rawValue;
+        requests[_getId(msg.sender, identifier, timestamp)] = Request({
+            proposer: address(0),
+            disputer: address(0),
+            currency: currency,
+            settled: false,
+            refundOnDispute: false,
+            proposedPrice: 0,
+            resolvedPrice: 0,
+            expirationTime: 0,
+            reward: reward,
+            finalFee: finalFee,
+            bond: finalFee,
+            customLiveness: 0
+        });
+
+        if (reward > 0) {
+            currency.safeTransferFrom(msg.sender, address(this), reward);
+        }
+
+        emit RequestPrice(msg.sender, identifier, timestamp, address(currency), reward, finalFee);
+
+        // The total bond is the final fee + the default bond, which is set to the final fee. So the total bond is
+        // 2x the final fee.
+        return finalFee.mul(2);
```

Comment:
Can the `default bond` get overwritten anywhere? or is that set on this contract deployment

Reply:
The default bond is the final fee. That defaulting logic is set on contract deployment. However, a contract that requests a price can customize the bond by calling `setBond()`. Do you think this should be otherwise changeable?

Reply:
Ah I see this is a big confusing. So the actual proposal bond is set in `propose()` by this line: `totalBond = request.bond.add(request.finalFee)`.

So basically this return value, defaulted and hardcoded to ` finalFee.mul(2)` is indicating to the caller that this is the default bond value. I think comments on this function specifically about the return value (e.g. "this function returns the initial proposal bond for this request, which can be changed by the requested of the price") would clear this up.

This makes sense to me now.

Reply:
Gotcha. +1 -- will add clearer comments!

Reply:
Done.